src/Microsoft.Extensions.DependencyModel/JsonTextReaderExtensions.cs

```diff
@@ -16,7 +16,19 @@ internal static bool TryReadStringProperty(this JsonTextReader reader, out strin
             if (reader.Read() && reader.TokenType == JsonToken.PropertyName)
             {
                 name = (string)reader.Value;
-                value = reader.ReadAsString();
+                
+                if (reader.Read())
+                {
+                    if (reader.TokenType == JsonToken.String)
+                    {
+                        value = (string)reader.Value;
+                    }
+                    else
+                    {
+                        reader.Skip();
```

Comment:
Yes, the `true` indicates that the caller should keep probing for more properties.